# HG changeset patch
# User Rik <octave@nomad.inbox5.com>
# Date 1301776672 25200
#      Sat Apr 02 13:37:52 2011 -0700
# Node ID 6ef23b4a34026cf72145ae20a7b7474c91c3f013
# Parent  b22816427ce92ae94439e0e42e7ca4c1cee26e48
Add quantile, prctile functions to documentation.

diff --git a/doc/ChangeLog b/doc/ChangeLog
--- a/doc/ChangeLog
+++ b/doc/ChangeLog
@@ -1,8 +1,13 @@
+2011-04-02  Rik  <octave@nomad.inbox5.com>
+
+	* interpreter/stats.txi: Add quantile, prctile functions to
+	documentation.
+
 2011-04-02  Rik  <octave@nomad.inbox5.com>
 
 	* interpreter/system.txi: Add isdeployed function to documentation.
 
 2011-03-31  Rik  <octave@nomad.inbox5.com>
 
 	* interpreter/io.txi: Add list_in_columns, terminal_size functions
 	to documentation.
diff --git a/doc/interpreter/stats.txi b/doc/interpreter/stats.txi
--- a/doc/interpreter/stats.txi
+++ b/doc/interpreter/stats.txi
@@ -91,16 +91,20 @@ measures to be developed.
 @DOCSTRING(var)
 
 @DOCSTRING(skewness)
 
 @DOCSTRING(kurtosis)
 
 @DOCSTRING(moment)
 
+@DOCSTRING(quantile)
+
+@DOCSTRING(prctile)
+
 A summary view of a data set can be generated quickly with the
 @code{statistics} function.
 
 @DOCSTRING(statistics)
 
 @node Basic Statistical Functions
 @section Basic Statistical Functions
 
diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,8 +1,13 @@
+2011-04-02  Rik  <octave@nomad.inbox5.com>
+
+	* statistics/base/prctile.m, statistics/base/quantile.m: Improve 
+	docstrings.
+
 2011-03-31  Rik  <octave@nomad.inbox5.com>
 
 	* pkg/module.mk, pkg/pkg.m, pkg/private/get_forge_pkg.m: Add
 	documentation for '-forge' option (bug #32464).  Make get_forge_pkg
 	a private function.
 
 2011-03-31  Marco Caliari  <marco.caliari@univr.it>
 
diff --git a/scripts/statistics/base/prctile.m b/scripts/statistics/base/prctile.m
--- a/scripts/statistics/base/prctile.m
+++ b/scripts/statistics/base/prctile.m
@@ -12,26 +12,28 @@
 ## MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 ## General Public License for more details.
 ##
 ## You should have received a copy of the GNU General Public License
 ## along with Octave; see the file COPYING.  If not, see
 ## <http://www.gnu.org/licenses/>.
 
 ## -*- texinfo -*-
-## @deftypefn  {Function File} {@var{y} =} prctile (@var{x}, @var{p})
+## @deftypefn  {Function File} {@var{q} =} prctile (@var{x})
+## @deftypefnx {Function File} {@var{q} =} prctile (@var{x}, @var{p})
 ## @deftypefnx {Function File} {@var{q} =} prctile (@var{x}, @var{p}, @var{dim})
-## For a sample @var{x}, compute the quantiles, @var{y}, corresponding
+## For a sample @var{x}, compute the quantiles, @var{q}, corresponding
 ## to the cumulative probability values, @var{p}, in percent.  All non-numeric
 ## values (NaNs) of @var{x} are ignored.
 ##
 ## If @var{x} is a matrix, compute the percentiles for each column and
 ## return them in a matrix, such that the i-th row of @var{y} contains the
 ## @var{p}(i)th percentiles of each column of @var{x}.
 ##
+## If @var{p} is unspecified, return the quantiles for @code{[0 25 50 75 100]}.
 ## The optional argument @var{dim} determines the dimension along which
 ## the percentiles are calculated.  If @var{dim} is omitted, and @var{x} is
 ## a vector or matrix, it defaults to 1 (column-wise quantiles).  When
 ## @var{x} is an N-d array, @var{dim} defaults to the first non-singleton
 ## dimension.
 ## @seealso{quantile}
 ## @end deftypefn
 
diff --git a/scripts/statistics/base/quantile.m b/scripts/statistics/base/quantile.m
--- a/scripts/statistics/base/quantile.m
+++ b/scripts/statistics/base/quantile.m
@@ -82,16 +82,26 @@
 ## S Language.  Wadsworth & Brooks/Cole.
 ##
 ## @item Hyndman, R. J. and Fan, Y. (1996) Sample quantiles in
 ## statistical packages, American Statistician, 50, 361--365.
 ##
 ## @item R: A Language and Environment for Statistical Computing;
 ## @url{http://cran.r-project.org/doc/manuals/fullrefman.pdf}.
 ## @end itemize
+##
+## Examples:
+##
+## @example
+## @group
+## x = randi (1000, [10, 1]);  # Create random empirical data in range 1-1000
+## q = quantile (x, [0, 1]);   # Return minimum, maximum of empirical distribution
+## q = quantile (x, [0.25 0.5 0.75]); # Return quartiles of empirical distribution
+## @end group
+## @end example
 ## @seealso{prctile}
 ## @end deftypefn
 
 ## Author: Ben Abbott <bpabbott@mac.com>
 ## Description: Matlab style quantile function of a discrete/continuous distribution
 
 function q = quantile (x, p, dim = 1, method = 5)
 
